### PR TITLE
Global phase

### PR DESCRIPTION
**Pull request by pehamTom**
Added global phases to certain gates (like RZ) during parsing.

### CODE REVIEW

File: src/zx/FunctionalityConstruction.cpp [hunk 1]
```diff
@@ -57,6 +58,7 @@ namespace zx {
 
     void FunctionalityConstruction::addCphase(ZXDiagram& diag, const PiRational& phase, Qubit ctrl, Qubit target,
                                               std::vector<Vertex>& qubits) {
+        // diag.addGlobalPhase(phase / 4);
```
Comment (burgholzer) [inline src/zx/FunctionalityConstruction.cpp]: left-over comment?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -53,7 +53,8 @@ package_add_test(${PROJECT_NAME}_test
         ${CMAKE_CURRENT_SOURCE_DIR}/unittests/test_qfr_functionality.cpp
         ${CMAKE_CURRENT_SOURCE_DIR}/algorithms/eval_dynamic_circuits.cpp
         ${CMAKE_CURRENT_SOURCE_DIR}/unittests/test_zx_functionality.cpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/unittests/test_dd_noise_functionality.cpp)
+        ${CMAKE_CURRENT_SOURCE_DIR}/unittests/test_dd_noise_functionality.cpp
+      )
```
Comment (burgholzer) [inline test/CMakeLists.txt]: intended format change?

File: test/unittests/test_zx_functionality.cpp [hunk 3]
```diff
@@ -120,7 +120,6 @@ TEST_F(ZXDiagramTest, complex_circuit) {
     zx::fullReduce(diag);
     EXPECT_EQ(diag.getNVertices(), 6);
     EXPECT_EQ(diag.getNEdges(), 3);
-    EXPECT_TRUE(diag.isIdentity());
```
Comment (burgholzer) [inline test/unittests/test_zx_functionality.cpp:123]: shouldn't this check be replaced with something appropriate?

File: test/unittests/test_zx_functionality.cpp [hunk 4]
```diff
@@ -201,3 +200,20 @@ TEST_F(ZXDiagramTest, InitialLayout) {
...
     EXPECT_TRUE(d.isIdentity());
 }
+
+TEST_F(ZXDiagramTest, RZ) {
+    qc = qc::QuantumComputation(2);
+    qc.rz(0, zx::PI / 8);
+
+    auto qcPrime = qc::QuantumComputation(2);
+    qcPrime.phase(1, zx::PI / 8);
+
```
Comment (burgholzer) [inline test/unittests/test_zx_functionality.cpp]: are these gates intentionally applied to different qubits? shouldn't there only be a single qubit in each of the circuits?

**Review by burgholzer — CHANGES_REQUESTED** [review]
Many thanks for this PR!

Just a couple of small things/questions:
 - why were the `json` and `pybind11` submodules touched in this PR?
 - just to double check: the `rx` gate does not need a global phase?
 - for the rest of the questions, see below.

File: src/zx/FunctionalityConstruction.cpp (hunk 1 above)
Comment (pehamTom) [reply src/zx/FunctionalityConstruction.cpp]: Yes :)

File: test/CMakeLists.txt (hunk 2 above)
Comment (pehamTom) [reply test/CMakeLists.txt]: Nope

File: test/unittests/test_zx_functionality.cpp (hunk 3 above)
Comment (pehamTom) [reply test/unittests/test_zx_functionality.cpp:123]: Definitely. I have adjusted this accordingly

File: test/unittests/test_zx_functionality.cpp (hunk 4 above)
Comment (pehamTom) [reply test/unittests/test_zx_functionality.cpp]: That is certainly an oversight on my part.

**Review by burgholzer — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**